elyra/pipeline/pipeline_definition.py

```diff
@@ -450,38 +506,18 @@ def propagate_pipeline_default_properties(self):
             if not pipeline_default_value:
                 continue
 
-            for pipeline in self.pipelines:
-                for node in pipeline.nodes:
-                    if Operation.is_generic_operation(node.op):
-                        node_value = node.get_component_parameter(pipeline_default_prop)
-                        if not node_value:
-                            node.set_component_parameter(pipeline_default_prop, pipeline_default_value)
-                        else:
-                            if pipeline_default_prop == pipeline_constants.ENV_VARIABLES:
-                                # Transform both into dicts
-                                pipeline_default_env_dict = self.envs_to_dict(env_list=pipeline_default_value)
-                                node_env_dict = self.envs_to_dict(env_list=node_value)
-                                merged_env_list = self.env_dict_to_list({**pipeline_default_env_dict, **node_env_dict})
-                                node.set_component_parameter(pipeline_default_prop, merged_env_list)
-
-    def envs_to_dict(self, env_list: List) -> Dict[str, str]:
-        envs = {}
-        for nv in env_list:
-            if nv:
-                if "=" not in nv:
-                    raise ValueError(f"Environmental variable {nv} does not contain an '=' assignment operator.")
-                else:
-                    nv_pair = nv.split("=", 1)
-                    if len(nv_pair) == 2 and nv_pair[0].strip():
-                        if len(nv_pair[1]) > 0:
-                            envs[nv_pair[0]] = nv_pair[1]
-        return envs
-
-    def env_dict_to_list(self, env_dict: Dict) -> List[str]:
-        envs = []
-        for env, env_value in env_dict.items():
-            envs.append(f"{env}={env_value}")
-        return envs
+            for node in self.pipeline_nodes:
+                if not Operation.is_generic_operation(node.op):
+                    continue
+
+                node_value = node.get_component_parameter(pipeline_default_prop)
+                if not node_value:
+                    node.set_component_parameter(pipeline_default_prop, pipeline_default_value)
+                    continue
+
+                if isinstance(pipeline_default_value, KeyValueList):
+                    merged_list = pipeline_default_value.merge_kv_pairs(node_value)
+                    node.set_component_parameter(pipeline_default_prop, merged_list)
```

Comment:
We should probably ensure that `node_value` is also an instance of `KeyValueList`.  If using the class factory approach described previously, this would become:
```suggestion
                if isinstance(pipeline_default_value, KeyValueList):
                    assert(isinstance(node_value, KeyValueList), "'node_value' is not of type KeyValueList!")
                    merged_list = KeyValueList.merge(node_value, pipeline_default_value)
                    node.set_component_parameter(pipeline_default_prop, merged_list)
```